var/spack/repos/builtin/packages/yaml-cpp/package.py

```diff
@@ -32,6 +37,23 @@ class YamlCpp(CMakePackage):
     conflicts('%intel@:11.1', when='@0.6.0:', msg="versions 0.6.0: require c++11 support")
     conflicts('%xl@:13.1', when='@0.6.0:', msg="versions 0.6.0: require c++11 support")
     conflicts('%xl_r@:13.1', when='@0.6.0:', msg="versions 0.6.0: require c++11 support")
+    conflicts('%clang cxxflags="-stdlib=libc++"', when='+tests',
+              msg=yaml_cpp_tests_libcxx_error_msg)
+
+    def flag_handler(self, name, flags):
+        # We cannot catch all conflicts with the conflicts directive because
+        # the user can add arbitrary strings to the flags. Here we can at least
+        # fail early.
+        # We'll include cppflags in case users mistakenly put c++ flags there.
+        spec = self.spec
+        if name in ('cxxflags', 'cppflags') and spec.satisfies('+tests'):
+            if '-stdlib=libc++' in flags:
+                raise ConflictsInSpecError(spec,
```

Comment:
get rid of the global `yaml_cpp_tests_libcxx_error_msg` and just end this line with `[`.  Then everything below will be much less indented.

Reply:
@tgamblin The global makes it so we don't have the same message copied into two places (the other copy is in the `conflicts` directive. I can fix the indentation without changing that.